### PR TITLE
🔧 Add vitest tests

### PR DESCRIPTION
**Pull request by DaniFoldi**

### CODE REVIEW

File: packages/comet/src/logger.ts [hunk 1]
```diff
@@ -27,6 +27,8 @@ export const logger: Logger = {
 }
 
 export function recordException(exception: unknown) {
+  console.error('[Comet] Recorded exception', exception)
```
Comment (andras-adam) [inline packages/comet/src/logger.ts:30]: Do we log to the console anywhere else?

File: packages/comet/src/error.ts [hunk 2]
```diff
@@ -73,6 +73,16 @@ export class CometErrorHandler {
             status: 400,
             headers: { 'content-type': 'application/json' }
           })
+        case ErrorType.InvalidCompatibilityDate:
+          return new Response(JSON.stringify({ success: false, error: 'Invalid compatibility date' }), {
```
Comment (andras-adam) [inline packages/comet/src/error.ts:77]: I'm not sure about including `{ success: false }`, a client can probably ascertain the success of their request by checking if the status code is 200-299, or just `Response.ok`

File: packages/example/test/api.ts [hunk 3]
```diff
@@ -0,0 +1,66 @@
...
+  const response = await SELF.fetch('https://domain/')
+  expect(response.status).toBe(404)
+})
+
+it ('should return 200 on GET /api/test', async () => {
+  const response = await SELF.fetch('https://domain/api/test')
+  expect(response.status).toBe(200)
+})
+
+// This should be 405, once comet supports that
```
Comment (andras-adam) [inline packages/example/test/api.ts:14]: I think it should now be supported on main

**Review by andras-adam — COMMENTED** [review]
Minor comments, and please rebase

File: packages/example/test/api.ts (hunk 3 above)
Comment (DaniFoldi) [reply packages/example/test/api.ts]: Nice! I've updated the test to cover it

File: packages/comet/src/logger.ts (hunk 1 above)
Comment (DaniFoldi) [reply packages/comet/src/logger.ts:30]: Hmm, it's all channeled into otel logs now.. should I remove this one too?

**Review by andras-adam — APPROVED** [review]

**Pull request merged**